### ISSUE TITLE
Loading TTF from string as image doesn't work (may also not in vips itself)

### ISSUE THREAD

**Issue by chregu** [libvips/php-vips#127]
The following fails with a Fatal Error in PHP

```
$v = \Jcupitt\Vips\Image::newFromBuffer(file_get_contents("Archivo-Regular.ttf"));
```

leads to 
```
PHP Fatal error:  Uncaught Jcupitt\Vips\Exception: magickload_buffer: Magick: improper image header `' @ error/tga.c/ReadTGAImage/222 (null)
```

but loading from file works

```
$v = \Jcupitt\Vips\Image::newFromFile("Archivo-Regular.ttf");
```

I can also reproduce it with `vips`

Does not work
```
$ cat Archivo-Regular.ttf | vips VipsCopy stdin  foo.png
VipsImage: unable to load source
```
Works:
```
vips VipsCopy Archivo-Regular.ttf foo.png
```

Also quickly checked with ImageMagick, that works there:

```
cat Archivo-Regular.ttf | convert - foo.png
```

It's really an edge case (because why you want to load a TTF as image ;)), but it works with files and I'm sure it worked with strings before.

**Comment by jcupitt** [comment]
Hi @chregu,

I think the problem is that TTF files look rather like TGA files. There's a sniffer here:

https://github.com/libvips/libvips/blob/master/libvips/foreign/magick.c#L55-L98

So your TTF is being detected as a TGA, and then the imagemagick TGA loader fails.

TGA isn't really an image format and has no unambigious header to sniff, so the detector tends to have false positives. The simplest fix would probably be to add a TTF sniffer just before the `if()` for TGA.

**Comment by jcupitt** [comment]
(we added the TGA sniffer recently, hence the regression you've hit)

**Comment by jcupitt** [comment]
... it was really easy, so I patched 8.12. Thanks for reporting this. Test prog:

```python
#!/usr/bin/python3

import sys
import pyvips

with open(sys.argv[1], "rb") as f:
    data = f.read()

image = pyvips.Image.magickload_buffer(data)

new_data = image.magicksave_buffer(format=sys.argv[2])
print(f"written {len(new_data)} bytes")
```

I now see:

```
$ ./try331.py ~/pics/MissFajardose-Regular.ttf PNG
written 58538 bytes
```

**Comment by chregu** [comment]
Thanks a lot. that indeed solved the issue
